options/rtdl/generic/linker.cpp

```diff
@@ -158,11 +214,14 @@ SharedObject *ObjectRepository::requestObjectWithName(frg::string_view name,
 
 SharedObject *ObjectRepository::requestObjectAtPath(frg::string_view path, uint64_t rts) {
 	// TODO: Support SONAME correctly.
-	auto it = _nameMap.get(path);
+	auto lastSlash = path.find_last('/') + 1;
+	auto name = path.sub_string(lastSlash, path.size() - lastSlash);
+	auto it = _nameMap.get(name);
 	if(it)
 		return *it;
 
-	auto object = frg::construct<SharedObject>(getAllocator(), path.data(), false, rts);
+	auto object = frg::construct<SharedObject>(getAllocator(),
+		name.data(), path.data(), false, rts);
```

Comment:
Path is used again below but maybe if we use `object->path` instead, we can move here?